lib/internal/errors.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+// The whole point behind this internal module is to allow Node.js to no
+// longer be forced to treat every error message change as a semver-major
+// change. The NodeError classes here all expose a `code` property whose
+// value statically and permanently identifies the error. While the error
+// message may change, the code should not.
+
+const kCode = Symbol('code');
+const kSymbolsMap = new Map();
+
+function symbolFor(name, add) {
+  var sym = kSymbolsMap.get(name);
+  if (add && sym === undefined) {
+    sym = Symbol(name);
+    kSymbolsMap.set(name, sym);
+  }
+  return sym;
+}
```

Comment:
I am not sure I understand why this file uses symbol properties of `exports`… wouldn’t `const messages = new Map();` as a key → message map be a bit easier?